### PR TITLE
feat: add email commands

### LINKED ISSUE #2: Add commands for email templates
There should be similar commands for working with email templates:
- `email:download` - Download all email templates to a directory
- `email:upload` - Upload all email templates from a directory
- `email:watch` - Watch a directory for changes

Handling for HTML / Text templates and localization must be implemented.

For consideration:
- `email:test` - Send a test email through FusionAuth to the given email address
- Should the base information of the email (Name, Default Subject, From Email, Default from name) also be exported / modifiable?

### PR DESCRIPTION
**Pull request by ColinFrick**
This PR adds the following email commands:
- `fusionauth email:create` - Create a new email template locally.
- `fusionauth email:download` - Download a specific template or all email templates from a FusionAuth server.
- `fusionauth email:duplicate` - Duplicate an email template locally.
- `fusionauth email:html-to-text` - Convert HTML email templates to text, where the text template is missing.
- `fusionauth email:upload` - Upload a specific template or all email templates to a FusionAuth server.
- `fusionauth email:watch` - Watch the email template directory and upload changes to a FusionAuth server.

The email directory is as follows:
```
emails
├── 004e5f2e-ec50-4eda-998e-c58afd2a048d
│   ├── body.html
│   ├── body.txt
│   ├── from_email.txt
│   ├── from_name.txt
│   ├── fuubar.txt
│   ├── name.txt
│   └── subject.txt
├── 530100f3-e25b-49e3-bdd0-46e4c2f06b9e
│   ├── de
│   │   ├── body.html
│   │   ├── body.txt
│   │   ├── from_name.txt
│   │   └── subject.txt
│   ├── fr
│   │   ├── body.html
│   │   ├── body.txt
│   │   ├── from_name.txt
│   │   └── subject.txt
│   ├── body.html
│   ├── body.txt
│   ├── from_email.txt
│   ├── from_name.txt
│   ├── name.txt
│   └── subject.txt
```

So we have a directory for each template containing the base properties and default templates. Locales are subdirectory containing the localized templates.

Closes #2

### CODE REVIEW

**Comment by mooreds** [comment]
@vcampitelli did you have a chance to look at this?

File: src/commands/email-download.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import {Command} from "commander";
+import {getEmailErrorMessage, getEmailSuccessMessage, reportError, validateEmailOptions} from "../utils.js";
+import {EmailTemplate, FusionAuthClient} from "@fusionauth/typescript-client";
+import * as fs from "fs";
```
Comment (vcampitelli) [inline src/commands/email-download.ts]:
```suggestion
import {existsSync} from "fs";
```

File: src/commands/email-download.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                emailTemplates = [clientResponse.response.emailTemplate];
+            } else {
+                emailTemplates = clientResponse.response.emailTemplates ?? []
+            }
+
+            for await (const emailTemplate of emailTemplates) {
+                const emailTemplateId = emailTemplate.id;
+                const emailTemplateDirectory = `${output}/${emailTemplateId}/`;
+
+                if (!fs.existsSync(emailTemplateDirectory)) {
```
Comment (vcampitelli) [inline src/commands/email-download.ts]:
```suggestion
                if (!existsSync(emailTemplateDirectory)) {
```

File: src/commands/email-download.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                await writeFile(`${emailTemplateDirectory}/body.html`, emailTemplate.defaultHtmlTemplate ?? '');
+                await writeFile(`${emailTemplateDirectory}/body.txt`, emailTemplate.defaultTextTemplate ?? '');
+                await writeFile(`${emailTemplateDirectory}/subject.txt`, emailTemplate.defaultSubject ?? '');
+                await writeFile(`${emailTemplateDirectory}/from_name.txt`, emailTemplate.defaultFromName ?? '');
+
+                // Export localized email templates
+                const locales = getLocalesFromEmailTemplates(emailTemplate);
+                for (const locale of locales) {
+                    const localeDirectory = `${emailTemplateDirectory}/${locale}/`;
+                    if (!fs.existsSync(localeDirectory)) {
```
Comment (vcampitelli) [inline src/commands/email-download.ts]:
```suggestion
                    if (!existsSync(localeDirectory)) {
```

File: src/commands/email-download.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            reportError(errorMessage, e);
+            process.exit(1);
+        }
+    });
+
+/**
+ * Gets the locales from a list of email templates
+ * @param emailTemplate
+ */
+export const getLocalesFromEmailTemplates = (emailTemplate: EmailTemplate): string[] => {
```
Comment (vcampitelli) [inline src/commands/email-download.ts]: Nice :sunglasses:

**Review by vcampitelli — CHANGES_REQUESTED** [review]
Just a few nitpicks. Great work though!

**Comment by mooreds** [comment]
@vcampitelli can you take another look, please?

**Comment by vcampitelli** [comment]
I pushed f1a38d94278af88ec41b8592a03947faa5ae761b to remove an `export` because the script was trying to add it as a command and it's just a helper function.

```
src/index.ts:23:63 - error TS2345: Argument of type 'Command<[], { output: string; locales: [] | string[]; }> | Command<[string | undefined], { output: string; }> | Command<[string | undefined], { output: string; key: string; host: string; clean: boolean; }> | ... 8 more ... | Command<...>' is not assignable to parameter of type 'CommandUnknownOpts'.
  Type '(emailTemplate: EmailTemplate) => string[]' is not assignable to type 'Command<unknown[], OptionValues>'.

23 Object.values(commands).forEach(command => program.addCommand(command));
```                                                                 ~~~~~~~

**Review by vcampitelli — APPROVED** [review]

**Pull request merged**